woocommerce/woocommerce
# Add To Cart Form - Stepper Layout: remove flex

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR removes the flex property from the form element. Although this change previously ensured the button matched the height of the input, it also caused multiple layout regressions in extensions.


| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/68cecbd9-026a-44de-8a05-9fb3c7abbe0c) | ![image](https://github.com/user-attachments/assets/2df9eb4d-b0f4-4c66-82a1-968af1600be4) | 


### With Product Add-Ons for WooCommerce

| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/c219455d-6d28-421b-a14b-c0b52eaf1ea2) | ![image](https://github.com/user-attachments/assets/5a5358d7-9ea3-4e39-bc8c-663758a6fd6d) |  


Closes #52488.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable TT4.
2. Edit the Single Product Template.
3. Ensure that you have the Add to Cart with Options block added.
4. Click on it.
5. Open the block settings.
6. Ensure that you have the `Quantity Selector` option
7. Ensure that the default one is `Input`.
8. Ensure that the input looks like this image:

![image](https://github.com/woocommerce/woocommerce/assets/4463174/a2263956-e591-4dc4-ac63-33695cb6d6ee)


10. Switch the option to `Stepper`
11. Ensure that the input looks like this image:
![image](https://github.com/woocommerce/woocommerce/assets/4463174/49585883-1798-4353-afb4-62083512d278)
**The following steps should be checked for both options and with a Simple Product, a Variation Product, and a Grouped Product**
12. Install [Product Add-Ons for WooCommerce extension](https://woocommerce.com/it/products/product-add-ons/).
13. Edit a Product.
14. Configure add-ons:
![image](https://github.com/user-attachments/assets/8432ba4e-9f16-45c4-97f8-2b566a878566)

15. Visit the Product.
16. Ensure that the input looks like the image in the table.
17. Ensure that the input works correctly (pay attention to focus/border/etc...).
18. Click Add to Cart.
19. Ensure that the right amount of the products is added with the right price.
20. Repeat with TT3 or another block theme.

### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add to Cart with Options - Stepper Layout: remove flex

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
